packages/jest-haste-map/src/index.js

```diff
@@ -125,7 +127,7 @@ class HasteMap {
 
     const list = options.providesModuleNodeModules;
     this._whitelist = (list && list.length)
-      ? new RegExp('(' + NODE_MODULES + '(?:' + list.join('|') + ')(?=$|' + path.sep + '))', 'g')
+      ? new RegExp(utils.replacePathSepForRegex('(' + NODE_MODULES + '(?:' + list.join('|') + ')(?=$|' + path.sep + '))'), 'g' )
```

Comment:
we have an 80-column-limit at FB (unfortunately not part of eslint :( )

I'd recommend breaking this up a bit, like:
```js
const nodeModules = utils.replacePathSepForRegex(NODE_MODULES);
const pathSep = utils.replacePathSepForRegex(path.sep);
this._whitelist = (list && list.length)
  ? new RegExp(…)
  : null;
```